File: src/CorporateMessengerWithAgile/Application/Dto/Summary/PositionInCompanySummaryDto.cs
namespace Application.Dto.Summary
{
    public class PositionInCompanySummaryDto : BaseDto
    {
        public Guid CompanyId { get; set; }
        public string Title { get; set; } = null!;
        public string Description { get; set; } = null!;
        public IReadOnlyList<Guid> EmployeeIds { get; set; } = null!;
    }
}

File: src/CorporateMessengerWithAgile/ReactApp/reactapp.client/src/models/entity/PositionInCompanySummaryDto.ts
// src/models/entity/PositionInCompanySummaryDto.ts
import { z } from 'zod';
import { BaseDtoSchema } from './BaseDto';
import { GuidSchema } from '../Guid';

export const PositionInCompanySummaryDtoSchema = BaseDtoSchema.extend({
    companyId: GuidSchema,
    title: z.string().min(1),
    description: z.string(),
    employeeIds: z.array(GuidSchema).optional(),
});

export type PositionInCompanySummaryDto = z.infer<typeof PositionInCompanySummaryDtoSchema>;
